### PR TITLE
#30 Database setup

### LINKED ISSUE #30: Setup a development database and integrate it into the kotlin project
**Is your feature request related to a problem? Please describe.**
We do not have a DB! Serious issue!

**Describe the solution you'd like**
I want to create a docker setup with a PostgeSQL database.
The DB will expose its ports.

**Describe alternatives you've considered**
We could also have a docker container which builds and serves the backend.
This could include auto refreshing

### PR DESCRIPTION
**Pull request by maxammann**
fixes #30

### CODE REVIEW

File: backend/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Properties to configure the gradle build
+
+exposed_version=0.25.1
```
Comment (RoAnBu) [inline backend/gradle.properties:3]: newest version is 0.28.1 on github, why version 25?

**Review by RoAnBu — APPROVED** [review]

File: backend/build.gradle.kts [hunk 2]
```diff
@@ -37,9 +44,18 @@ dependencies {
 
     // Use the Kotlin JUnit integration.
     testImplementation("org.jetbrains.kotlin:kotlin-test-junit")
+    
+    implementation("org.jetbrains.exposed", "exposed-core", exposed_version)
+    implementation("org.jetbrains.exposed", "exposed-dao", exposed_version)
+    implementation("org.jetbrains.exposed", "exposed-jdbc", exposed_version)
+    implementation("com.impossibl.pgjdbc-ng", "pgjdbc-ng", "0.8.6")
```
[the remark this answers is not in the document]
Comment (maxammann) [reply backend/build.gradle.kts:51]: This was recommended by exposed so I just went with it :D

File: docs/setup.md [hunk 3]
```diff
@@ -32,3 +32,18 @@ You have to add a file named `secrets.json` next to `pubspec.yaml` with content
...
+
+   The credentials are:
+
+   |Property|Value|
+   |---|---|
+   |Host (within Docker)|db|
+   |Username|postgres|
+   |Password|postgres|
+   |Database|ehrenamtskarte|
+4. Install JDK8
```
[the remark this answers is not in the document]
Comment (maxammann) [reply docs/setup.md:48]: Dunno :D

File: backend/gradle.properties (hunk 1 above)
Comment (maxammann) [reply backend/gradle.properties:3]: good catch!

File: backend/settings.gradle.kts [hunk 4]
```diff
@@ -8,3 +8,5 @@
  */
 
 rootProject.name = "backend"
+
```
[the remark this answers is not in the document]
Comment (maxammann) [reply backend/settings.gradle.kts:11]: nope

**Pull request merged**